streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -261,7 +293,7 @@ public String toString() {
      * @throws IllegalStateException if instance is currently running
      */
     public void cleanUp() {
-        if (state == RUNNING) {
+        if (state == KafkaStreamsState.RUNNING || state == KafkaStreamsState.REBALANCING) {
```

Comment:
AFAICT we are not testing (before and after this PR) whether this exception is actually being thrown.